sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsServiceSerializer.cs

```diff
@@ -0,0 +1,776 @@
+            }
+            json.WriteEndArray();
+            json.WriteEndObject();
+            json.Flush();
+            return writer.WrittenMemory;
+        }
+
+        #endregion Serialize Inputs
+
+        #region Deserialize Common
+
+        private static string ReadDocumentId(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("id", out JsonElement idValue))
+                return idValue.ToString();
+
+            return default;
+        }
+
+        private static DocumentStatistics ReadDocumentStatistics(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("statistics", out JsonElement statisticsValue))
+            {
+                DocumentStatistics statistics = new DocumentStatistics();
+
+                if (statisticsValue.TryGetProperty("charactersCount", out JsonElement characterCountValue))
+                    statistics.CharacterCount = characterCountValue.GetInt32();
+                if (statisticsValue.TryGetProperty("transactionsCount", out JsonElement transactionCountValue))
+                    statistics.TransactionCount = transactionCountValue.GetInt32();
+
+                return statistics;
+            }
+
+            return default;
+        }
+
+        private static void ReadDocumentErrors(JsonElement documentElement, List<DocumentError> errors)
+        {
+            if (documentElement.TryGetProperty("errors", out JsonElement errorsValue))
+            {
+                foreach (JsonElement errorElement in errorsValue.EnumerateArray())
+                {
+                    DocumentError error = new DocumentError();
+
+                    if (errorElement.TryGetProperty("id", out JsonElement idValue))
+                        error.Id = idValue.ToString();
+                    if (errorElement.TryGetProperty("error", out JsonElement errorValue))
+                    {
+                        if (errorsValue.TryGetProperty("message", out JsonElement messageValue))
+                        {
+                            error.Message = messageValue.ToString();
+                        }
+                    }
+
+                    errors.Add(error);
+                }
+            }
+        }
+
+        private static string ReadModelVersion(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("modelVersion", out JsonElement modelVersionValue))
+            {
+                return modelVersionValue.ToString();
+            }
+
+            return default;
+        }
+
+        private static DocumentBatchStatistics ReadDocumentBatchStatistics(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("statistics", out JsonElement statisticsElement))
+            {
+                DocumentBatchStatistics statistics = new DocumentBatchStatistics();
+
+                if (statisticsElement.TryGetProperty("documentsCount", out JsonElement documentCountValue))
```

Comment:
Each `TryGetProperty` is O(n). I recommend a loop instead.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8783